src/components/shared/SpentTime/SpentTime.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { useCallback, useEffect, useMemo, useState } from "react";
+import { useFirestore } from "reactfire";
+import { collection, getDocs, query, where } from "firebase/firestore";
+import { chunk } from "lodash";
+
+import {
+  COLLECTION_SPACES,
+  FIRESTORE_QUERY_IN_ARRAY_MAX_ITEMS,
+} from "settings";
+
+import { getUserRef } from "api/profile";
+
+import { UserVisit } from "types/Firestore";
+import { UserId } from "types/id";
+import { AnyVenue } from "types/venues";
+
+import { WithId } from "utils/id";
+import { isTruthy } from "utils/types";
+
+import { useWorldAndSpaceByParams } from "hooks/spaces/useWorldAndSpaceByParams";
+
+import styles from "./SpentTime.module.scss";
+
+interface SpentTimeProps {
+  userId: UserId;
+}
+
+export const SpentTime: React.FC<SpentTimeProps> = ({ userId }) => {
+  const { isLoaded: isWorldLoaded, spaceId } = useWorldAndSpaceByParams();
+  const [visits, setVisits] = useState<WithId<UserVisit>[]>([]);
+  const [venues, setVenues] = useState<WithId<AnyVenue>[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+
+  const firestore = useFirestore();
+
+  const fetchAllVenues = useCallback(async () => {
+    const userSnapshot = await getUserRef(userId).get();
+    const visitsSnapshot = await userSnapshot.ref.collection("visits").get();
```

Comment:
if `getUserRef()` returns the reference, there is no need for backtracking by `userSnapshot.ref`. It can be

```suggestion
    const userRef = getUserRef(userId);
    const userSnapshot = await userRef.get();
    const visitsSnapshot = await userRef.collection("visits").get();
```

Reply:
you're right, the logic could be simplified a bit, made it one-line. Done here: https://github.com/sparkletown/sparkle/pull/2953/commits/a52527d7fce74b5f15447723260cc7715c8614b3